### ISSUE TITLE
I am kind of doubt if this model could be called as a GAN.

### ISSUE THREAD

**Issue by ANYMS-A** [leftthomas/SEGAN#10]
Hi there, recently I'm trying to reproduce this SEGAN model and find out some questions.

1. The biggest question is about the loss function of the discriminator. As we know the original GAN's discriminator is doing binary classification task. So it use a Sigmoid at the last output layer and Binary Cross Entropy as the loss function. For this model's discriminator it seems it is doing a regression task, the loss function is trying to minimize the distance between outputs and 1 (or 0). So I think the discriminator contributes nothing to the final performance.  minimizing  L1 loss between clean speech and generated speech make the whole system work.

2. So I discarded the discriminator and only train the generator for speech enhancement, it gives a very close performance of SEGAN. If only use the generator for training, the model could be seen as a de-noising auto encoder.

3.I'm kind of confused about that how much does the discriminator contribute during the Adversarial Process.

Many thanks!

**Comment by leftthomas** [comment]
@crissallan you should contact with the author of that paper, I just reimplement SEGAN according to the paper
